### PR TITLE
Release Process Documentation 

### PR DESCRIPTION
**Pull request by kris-nova**
Documentation on the release process

Note: This is just a starting point, ideally I could get some feedback from @justinsb and @chrislovecnm as we get closer to doomsday

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1053)

### CODE REVIEW

**Comment by kris-nova** [comment]
@k8s-bot aws e2e test this

File: docs/releases.md [hunk 1]
```diff
@@ -1,4 +1,79 @@
...
+#### 1. Maintainers quorum
+
+* A majority of maintainers must agree on a single commit SHA for the release. The agreement should be documented in the release notes.
+* The maintainers will match the commit SHA to the release `major.minor.patch` in the release notes
+* The maintainers will agree on a changelog proposal that MUST be in a PR prior to the release.
+* The main `REAMDE.md` must also reflect the new release. Ideally this should be in the same PR.
+
+#### 2. Merge the release notes
+
+* `HISTORY.md` serves as our official changelog.
```
Comment (chrislovecnm) [inline docs/releases.md]: 👍

File: docs/releases.md [hunk 2]
```diff
@@ -1,4 +1,79 @@
...
+* Before the release, the maintainers had opened a PR with the proposed changelog. This should now be merged.
+
+
+#### 3. Release kops to github
+
+*  Darwin binary (generated)
+*  Linux binary (generated)
+*  Source code (zip)
+*  Source code (tar.gz)
```
Comment (chrislovecnm) [inline docs/releases.md]: Need to list:

- ami
- nodeup in s3
- kops in s3
- protokube docker
- dnscontroller docker
- channels <- this needs to pull a tag, not master

File: docs/releases.md [hunk 3]
```diff
@@ -1,4 +1,79 @@
...
+
+#### 4. Release kops to homebrew
+
+* Following the [documentation](development/homebrew.md) we must release a compatible homebrew formulae with the release.
+* This should be done at the same time as the release, and we will iterate on how to improve timing of this.
```
Comment (chrislovecnm) [inline docs/releases.md]: The gent that did homebrew is responsible for this.

File: docs/releases.md [hunk 4]
```diff
@@ -1,4 +1,79 @@
...
+
+#### 5. AWS support
+
+* Build and release new nodeup binary to S3
+* Verify/Validate all AMIs are updated
+
+#### 6. Update release branch
+
+* Merge the `master` branch into the `release` branch. [More information](releases.md#branch-strategy)
```
Comment (chrislovecnm) [inline docs/releases.md]: We need to tag a release out of the release branch.

File: docs/releases.md [hunk 5]
```diff
@@ -1,4 +1,79 @@
...
+
+#### 7. Manual test and validate
+
+* Maintainers should now give the repository a once over to validate everything looks in place.
+* At least 1 maintainer must run the recently released code to verify success.
```
Comment (chrislovecnm) [inline docs/releases.md]: 2?

Comment (kris-nova) [inline docs/releases.md]: Fixed.

File: docs/releases.md (hunk 4 above)
Comment (kris-nova) [inline docs/releases.md]: Fixed.

File: docs/releases.md (hunk 3 above)
Comment (kris-nova) [inline docs/releases.md]: I think ultimately `kops` maintainers are responsible for this, we can push on him to help, but at the end of the day it's on us to make sure this stay updated. If we are eventually scripting this process, we need it here.

File: docs/releases.md (hunk 2 above)
Comment (kris-nova) [inline docs/releases.md]: Fixed  (See AWS Support)

**Comment by kris-nova** [comment]
@chrislovecnm - Your concerns have been addressed

**Comment by kris-nova** [comment]
@k8s-bot aws e2e test this

**Comment by kris-nova** [comment]
Getting this in so I can start the next PR for `1.4.2` - I know it's bad practice to merge my own PRs - but I would like this in for the new fork.. if there are outstanding issues comment here as you normally would and I will port them over in the next PR.. I will add a pointer as well

Cheers

**Pull request merged**